### ISSUE TITLE
Idea about Mod compatibility with Fabric Seasons

### ISSUE THREAD

**Issue by sternschnaube** [rfresh2/XaeroPlus#97]
https://modrinth.com/mod/fabric-seasons

Using this mod and putting the cooldown down leads to alltime map refreshes.
My idea would be, add another 4 map types to the normal version.

If the mod with a season is detected, it switches to the season map from it. With this, you avoid countless map refreshes 😄 

I like to hear your opinion about this. Keep up the great work with XaeroPlus 🥳

**Comment by rfresh2** [comment]
I understand the problem but the solution isn't simple. There aren't a concept of "map types" in xaero's. You can manually switch between multiple maps per server but only one is active at a time (i.e. data is actually written and read).

There might be a way to store and render the block colors dynamically based on the season but it would be a bit tricky to do.

**Comment by sternschnaube** [comment]
Do you think that multiple map switching could be the way? That only one is active is absolutely fine - but automatical switching would be great. Option to switch it to manual.

Your second point sounds interesting - it would also reduce (a lot) storage, since you have only one map.

**Comment by rfresh2** [comment]
> Do you think that multiple map switching could be the way

Nope. It wouldn't be a very good user experience if the map is constantly missing chunks whenever seasons change.

**Comment by sternschnaube** [comment]
You have to run and unlock the areas multiple times - but that's something the user should decide in the end. It would be at least a better behaviour than multiple refreshes from the main map.

A way to convert / save the data dynamically would be the best.

**Comment by sternschnaube** [comment]
Any update on this @rfresh2? 😊 
Something simple like multiple maps and just switching between them, at the cost of more storage would be already good enough.

**Comment by rfresh2** [comment]
there is actually a better possible solution now

![Image](https://github.com/user-attachments/assets/7a811bcf-cd2f-4aca-b694-1c78cd49d7a8)

this can take awhile for large maps but it should refresh all the colors

you can test it by manually clicking that when your season changes if you want

**Comment by sternschnaube** [comment]
Wow! That's a cheap way, and it's more than enough currently. Thank you for telling me about this feature.
I stick with finished map and update if needed.
